flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -564,6 +565,36 @@ public boolean createView(String path, Table view, boolean ignoreIfExists) {
         return catalogManager.createTable(tableTable, viewIdentifier, ignoreIfExists);
     }
 
+    @Override
+    public void createModel(String path, ModelDescriptor descriptor, boolean ignoreIfExists) {
+        Preconditions.checkNotNull(path, "Path must not be null.");
+        Preconditions.checkNotNull(descriptor, "Model descriptor must not be null.");
+        final ObjectIdentifier objectIdentifier =
+                catalogManager.qualifyIdentifier(getParser().parseIdentifier(path));
+        catalogManager.createModel(descriptor.toCatalogModel(), objectIdentifier, ignoreIfExists);
+    }
+
+    @Override
+    public void createModel(String path, ModelDescriptor descriptor) {
+        createModel(path, descriptor, false);
+    }
+
+    @Override
+    public void createTemporaryModel(String path, ModelDescriptor descriptor) {
+        createTemporaryModel(path, descriptor, false);
+    }
+
+    @Override
+    public void createTemporaryModel(
+            String path, ModelDescriptor descriptor, boolean ignoreIfExists) {
+        Preconditions.checkNotNull(path, "Path must not be null.");
+        Preconditions.checkNotNull(descriptor, "Model descriptor must not be null.");
+        final ObjectIdentifier objectIdentifier =
+                catalogManager.qualifyIdentifier(getParser().parseIdentifier(path));
```

Comment:
nit: could be extracted into a separate private method with replacement all occurrences